backend/internal/middleware/auth_test.go

```
@@ -0,0 +1,208 @@
+					return
+				}
+
+				token := authHeader[len(bearerPrefix):]
+				user, err := authSvc.ValidateToken(token)
+				if err != nil {
+					http.Error(w, "invalid token", http.StatusUnauthorized)
+					return
+				}
+
+				// Add user to context
+				ctx := context.WithValue(r.Context(), "user", user)
```

golangci-lint (1.64.8)

SA1029: should not use built-in type string as key for value; define your own type to avoid collisions

(staticcheck)